pkg/storage/unified/resource/server.go

```diff
@@ -598,6 +605,10 @@ func (s *server) Watch(req *WatchRequest, srv ResourceStore_WatchServer) error {
 				}); err != nil {
 					return err
 				}
+
+				// record latency
+				latencyMilliseconds := time.Now().UnixMicro() - event.ResourceVersion
+				StorageServerMetrics.WatchEventLatency.WithLabelValues(event.Key.Resource).Observe(float64(latencyMilliseconds))
```

Comment:
What's your take regarding measuring on the server side vs client side ? We could also measure this in the broadcaster on the server (there is only one per server) which isn't affected by slow readers. We could measure on the client side (in the apistore/store Storage interface). Here the measurement is on the Watch function which is called by by every watcher.

Reply:
I like recording it on the server because if it's slow, it's in an area we're 100% responsible for, and so the alert can be more actionable for our team. If we record it on the client, there may be other things affecting latency out of our control. Knowing the latency for the client would be helpful though.

I wouldn't mind recording both. Would make it easier to see where any latency is coming from. Thoughts?

Reply:
Server side makes sense !